backend/src/main/scala/com/softwaremill/adopttapir/starter/StarterService.scala

```diff
@@ -2,17 +2,60 @@ package com.softwaremill.adopttapir.starter
 
 import cats.effect.IO
 import com.softwaremill.adopttapir.logging.FLogging
+import com.softwaremill.adopttapir.template.{FileTemplate, ProjectTemplate}
 import com.softwaremill.adopttapir.util.Clock
 
 import java.io.File
+import java.util.zip.Deflater
+import scala.reflect.io.Directory
 
 class StarterService(
     clock: Clock,
-    config: StarterConfig
+    config: StarterConfig,
+    template: ProjectTemplate
 ) extends FLogging {
+
   def generateZipFile(starterDetails: StarterDetails): IO[File] = {
-    IO {
-      new java.io.File("src/main/resources/temporal.zip")
+    val filesToCreate = prepareTemplateFiles(starterDetails)
+    IO(os.temp.dir(prefix = config.tempPrefix).toIO)
+      .bracket { tempDir =>
+        for{
+          _ <- logger.debug("created temp dir: " + tempDir.toString)
+          _ <- storeFiles(tempDir, filesToCreate)
```

Comment:
maybe we can do the cleanup:
* in a non-typesafe way, in an interceptor, checking if the request is completed using the zip-serving endpoint, and attaching the cleanup using the `BodyListener`
* in a typesafe way, by returning a stream from the endpoint instead of a file, and attaching the cleanup in the stream's `onComplete`

Reply:
Done with usage of stream approach. Thanks for hint.